core/include/traccc/fitting/kalman_filter/gain_matrix_updater.hpp

```diff
@@ -45,13 +45,18 @@ struct gain_matrix_updater {
 
         const auto D = trk_state.get_measurement().meas_dim;
         assert(D == 1u || D == 2u);
-        if (D == 1u) {
-            return update<1u, shape_type>(trk_state, bound_params);
-        } else if (D == 2u) {
-            return update<2u, shape_type>(trk_state, bound_params);
+        bool result = false;
+        switch (D) {
+            case 1u:
+                result = update<1u, shape_type>(trk_state, bound_params);
+                break;
+            case 2u:
+                result = update<2u, shape_type>(trk_state, bound_params);
+                break;
```

Comment:
```suggestion
            case 1u:
                return update<1u, shape_type>(trk_state, bound_params);
            case 2u:
                return update<2u, shape_type>(trk_state, bound_params);
```

Reply:
The reason I went with that setup was two-fold:
  - I'm not 100% sure what happens with `__builtin_unreachable()` on all the platforms that this code will run on. 

I was considering putting

```c++
default:
    __builtin_unreachable();
    return false;
```

there, "just to be safe". But this formalism seemed better.

  - GPU functions can sometimes have hangups about returning at different points. Many coding rules even forbid multiple return points from functions. So I thought this may actually help.

But admittedly, I'm not absolutely sure about either of these points. :thinking: